docs/architecture/modernize-desktop/why-modern-desktop-applications.md

```diff
@@ -0,0 +1,176 @@
+
+Microsoft's approach to modernizing existing applications is to give you the flexibility to create your own customized path. All the modernization strategies described in this book are mostly independent. You can choose ones that are relevant for your application and skip others that aren't important for you. In other words, you can mix and match the strategies to best address your application needs.
+
+## Desktop applications nowadays
+
+Before the raise of the Internet, desktop applications were the main approach to build software systems. Developers could choose any programming language, such as COBOL, Fortran, VB6, or C++. But where they developed small tools or complex distributed architectures, they were all desktop applications.
+
+Then, Internet technologies started shocking the development world and winning over more and more engineers with advantages like easy deployment and simplified distribution processes. The fact that once a web application was deployed to production all users got automatic updates made a huge impact on the software agility.
+
+However, the Internet infrastructure, underlying protocols, and standards like HTTP and HTML weren't designed for building complex applications. In fact, the major development effort back then was aiming just one goal: to give web applications same capabilities that desktop applications have, such as fast data input and state management.
+
+Even though web and mobile applications have grown at an incredible pace, for certain tasks desktop applications still hold the number one place in terms of efficiency and performance. That explains why there are millions of developers who are building their projects with WPF and WinForms and the amount of those applications is constantly growing.
+
+Here are some reasons for choosing desktop applications in your development:
+
+- Desktop apps have better interaction with user's PC.
+- The performance of desktop applications for complex calculations is much higher than performance of web applications.
+- Running custom logic on the client side is possible but much harder with a web application.
+- Using multithreading is easier and more efficient in a desktop application.
+- The learning curve for designing user interfaces (UIs) isn't steep. And for WinForms, it's completely intuitive with drag-and-drop experience of the Windows Forms designer.
+- It's easy to start coding and testing your algorithms without the need to set up a server infrastructure or to care about connectivity problems, firewalls, and browser compatibility.
+- Debugging is powerful as compared to web debugging.
+- Access to hardware devices, such as camera, Bluetooth, or card readers, is easy.
+- Since the technology has been around for a while, there's a large number of experts and a knowledge base available to develop desktop applications.
```

Comment:
```suggestion
- Since the technology has been around for a while, there are many experts and a knowledge base available to develop desktop applications.
```